### DISCUSSION TITLE
make lobotomize not affect villagers that can still sleep

### DISCUSSION THREAD

**Discussion by hirata-estate** [PurpurMC/Purpur#1010] [__Dismissed__] [answered]
The lobotomize villagers feature is great, especially for survival servers like mine with players that _love_ to make huge laggy trading halls. The issue is, as shown in the image here, enabling lobotomy breaks some designs of iron farms.
![image](https://user-images.githubusercontent.com/61897277/174757373-b4c4e632-9dfa-4ea7-a3da-977feccf079d.png)
It seems that it lobotomizes the villagers and they don't sleep, breaking some designs.

**Comment by encode42** [comment, marked as answer]
This is working as intended. Lobotomization doesn't puppet logic, such as sleeping, for the sake of performance. It'd be counter-intuitive if lobotomizing a villager is more expensive than just ticking villagers.

It's easy to adapt these iron farm designs to give the villagers an extra block of space to move around in. This way, the villagers aren't lobotomized, and the logic required for the farm is executed. I've seen a few modifications to popular iron farm designs in our Discord.
